granulate_utils/linux/cgroups/base_cgroup.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+import os
+from enum import Enum
+from pathlib import Path
+from typing import List
+
+from granulate_utils.linux.cgroups.exceptions import MissingCgroup, MissingController, SkippedCgroup, UnsupportedCgroup
+
+PID_CGROUPS = Path("/proc/self/cgroup")
+CGROUP_PARENT_PATH = Path("/sys/fs/cgroup")
+
+
+class HIERARCHIES(Enum):
+    memory = "memory"
+    cpu = "cpu,cpuacct"
+
+
+class CgroupVerifications:
```

Comment:
The method names here are not imperative and its hard to understand what they mean. Either -
1. Change to imperative names, i.e `supported_cgroup` -> `is_cgroup_controller_supported`, `cgroup_v1` -> `is_cgroup_v1` etc.
2. Coalesce all functions into one `verify_preconditions` function